test/sequential/sequencerStaking.rules.test.ts

```diff
@@ -381,6 +393,150 @@ describe("sequencerStaking", () => {
+    expect(maintenanceStatus.isMaintenance.toString()).toEqual("false");
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    const update = new L2Update(api)
+      .withDeposit(
+        txIndex,
+        sequencer.keyRingPair.address,
+        sequencer.keyRingPair.address,
+        BN_HUNDRED,
+      )
+      .withCancelResolution(txIndex, 1, true)
+      .on(chain)
+      .buildUnsafe();
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update),
+    ).then(async (events) => {
+      expectMGAExtrinsicSuDidSuccess(events);
+    });
+    const event = await waitForEvents(api, "rolldown.RequestProcessedOnL2", 40);
+    const depositErr = await getEventError(event);
+    const resolutionErr = await getEventError(event, 1);
+    expect(depositErr).toEqual(undefined);
+    expect(resolutionErr).toEqual("WrongCancelRequestId");
+    maintenanceStatus = await api.query.maintenance.maintenanceStatus();
+    expect(maintenanceStatus.isMaintenance.toString()).toEqual("true");
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        foundationMembers[0],
+        Maintenance.switchMaintenanceModeOff(),
+      ),
+    );
+  });
+
+  it("When we have a failed deposit and send it again, it will result in no-execution again", async () => {
+    const chain = "Ethereum";
+    const sequencer = await setupASequencer(chain);
+    const api = getApi();
+    const txIndex = await Rolldown.lastProcessedRequestOnL2(chain);
+    const update1 = new L2Update(api)
+      .withDeposit(
+        txIndex,
+        sequencer.keyRingPair.address,
+        sequencer.keyRingPair.address,
+        BN_TEN_THOUSAND.pow(new BN(18)),
+      )
+      .on(chain)
+      .buildUnsafe();
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update1),
+    );
+    const event1 = await waitForEvents(
+      api,
+      "rolldown.RequestProcessedOnL2",
+      40,
+    );
+    const error1 = await getEventError(event1);
+    expect(error1).toEqual("Overflow");
+    const update2 = new L2Update(api)
+      .withDeposit(
+        txIndex,
+        sequencer.keyRingPair.address,
+        sequencer.keyRingPair.address,
+        BN_TEN_THOUSAND.pow(new BN(18)),
+      )
+      .withDeposit(
+        txIndex + 1,
+        sequencer.keyRingPair.address,
+        sequencer.keyRingPair.address,
+        BN_MILLION,
+      )
+      .on(chain)
+      .buildUnsafe();
+    await waitForSessionChange();
+    await waitNewBlock();
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update2),
+    );
+    const event2: any = await waitForEvents(api, "tokens.Endowed", 40);
```

Comment:
There is a waiter for the update to be executed, please take a look and ping me if you do not find it.